topics.py

```diff
@@ -0,0 +1,74 @@
+
+def compare_pubs(topic, mqtt_topic_prefix, target_device):
+    if topic == '{device_id}/shadow/get/accepted':
+        topic = target_device +'/shadow/get/accepted'
+    elif topic == '$aws/things/{device_id}/shadow/get/rejected':
+        topic = '$aws/things/' + target_device + '/shadow/get/rejected'
+    elif topic == '$aws/things/{device_id}/shadow/update/delta':
+        topic = '$aws/things/', target_device + '/shadow/update/delta'
+    elif topic == '{mqtt_topic_prefix}/m/d/{device_id}/d2c':
+        topic = mqtt_topic_prefix + 'm/d/' + target_device + '/d2c'
+    elif topic == '{mqtt_topic_prefix}/m/d/{device_id}/c2d':
+        topic = mqtt_topic_prefix + 'm/d/'+ target_device + '/c2d'
+    elif topic == '{mqtt_topic_prefix}/{device_id}/jobs/rcv':
+        topic = mqtt_topic_prefix + target_device + '/jobs/rcv'
+    elif topic == 'm/#':
+        topic = mqtt_topic_prefix + 'm/#'
+    elif topic == 'a/connections':
+        topic = mqtt_topic_prefix + 'a/connections'
+    else:
+        topic = topic  #user manual input
+    return topic
+
+def compare_subs(topic, mqtt_topic_prefix, target_device):
+    if topic == '$aws/things/{device_id}/shadow/get':
+        topic = '$aws/things/' + target_device + '/shadow/get'
```

Comment:
you can use the str.format() function to replace keywords in braces:
'{topic_prefix}/m/d/{dev_id}/d2c'.format(topic_prefix = mqtt_topic_prefix, dev_id = device_id)

which would simplify this a lot